youtube_dl/extractor/facebook.py

```diff
@@ -455,6 +456,13 @@ def _real_extract(self, url):
         if info_dict:
             return info_dict
 
+        # In case the url is a redirect and contains an external link.
+        if video_id.startswith("https://"):
+            return {
+                        '_type': 'url',
+                        'url': video_id,
+                    }
```

Comment:
Must be separate extractor.

Reply:
So a separate new extractor with a regex for the Facebook 'warning' URL? Any suggestions for the name for this new extractor? I was going to go with facebookWarning extractor.

Also, in the new extractor, won't we be ultimately passing the redirected URL to the generic.py, as the code is doing here?

Reply:
Name essentially does not matter. `FacebookExternalLinkIE` is a reasonable name. "The code here" does not pass to the generic extractor in the first place.

Reply:
I'm sorry, I meant 'common.py'. Then would that be fine to pass the link to common.py for download?

Reply:
It's already fine. More idiomatic way is to use `url_result`.

Reply:
Created another PR (#19116) with the correct branch as per the developer instructions.